File: src/main/java/me/diax/diax/commands/information/Suggest.java
package me.diax.diax.commands.information;

import me.diax.comportment.jdacommand.Command;
import me.diax.comportment.jdacommand.CommandDescription;
import net.dv8tion.jda.core.entities.Message;
import net.dv8tion.jda.core.JDA;

import java.util.Random;

@CommandDescription(
    name = "suggest",
    triggers = "suggest"
)
public class Suggest implements Command {

    private JDA jda;

    public Suggest(JDA jda) {
      this.jda = jda;
    }

    @Override
    public void execute(Message message, String args) {
        if (args.isEmpty()) {
            message.getTextChannel().sendMessage("Please give something to suggest").queue();
            return;
        }
        sendSuggestion(args);
    }
    
    private void sendSuggestion(String s) {
      jda.getTextChannelById("356564127864586240").sendMessage("New Suggestion: " + s).queue(); //sends message to #suggestions in diax server
    }
}

File: src/main/kotlin/me/diax/diax/commands/information/Suggest.kt
package me.diax.diax.commands.information

import me.diax.comportment.jdacommand.Command
import me.diax.comportment.jdacommand.CommandDescription
import me.diax.diax.util.Emote.SMILE
import me.diax.diax.util.StringUtil
import me.diax.diax.util.WebHookUtil
import me.diax.diax.util.style.Embed
import net.dv8tion.jda.core.entities.Message

@CommandDescription(
        name = "suggest",
        description = "[description]",
        triggers = ["suggest"]
)
class Suggest : Command {

    override fun execute(message: Message, s: String) {
        if (s.isBlank()) {
            message.channel.sendMessage(Embed.error("Please describe your suggestion!")).queue()
            return
        }
        WebHookUtil.suggest(message.jda, "```${StringUtil.stripMarkdown(s)}```\n*Suggested by ${StringUtil.stripMarkdown(message.author.name)}#${message.author.discriminator}*")
        message.channel.sendMessage("$SMILE - Your suggestion has been submitted, join here to track it: https://discord.gg/5sJZa2y").queue()
    }
}
